activerecord/lib/active_record/transactions.rb

```diff
@@ -209,6 +209,10 @@ def transaction(**options, &block)
         connection.transaction(**options, &block)
       end
 
+      def transactional_job_enqueue(*args, &block)
```

Comment:
Why is this method necessary? I see no reason to a new method to the public API to do this.

Reply:
I am just sort of spit balling this, I was not really sure where this should be or how the api should work so I put it here to get conversation going.